src/main/scala/is/hail/expr/Type.scala

```diff
@@ -2202,14 +2202,27 @@ final case class TStruct(fields: IndexedSeq[Field], override val required: Boole
     (t, selectF)
   }
 
+  var missingIdx: Array[Int] = _
```

Comment:
I kind of don't like this idiom.  It got used below because we needed to override `byteSize`.  I was looking for another way to implement it.  In this case I think you can just do:

```
  val (missingIdx, nMissingBytes) = {
    ...
    (a, (i + 7) >>> 3)
  }
```